moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -467,28 +494,22 @@ trajectory_msgs::JointTrajectory JogCalcs::composeJointTrajMessage(sensor_msgs::
     std::vector<double> acceleration(num_joints_);
     point.accelerations = acceleration;
   }
-  new_joint_traj.points.push_back(point);
-
-  return new_joint_traj;
+  joint_trajectory.points.push_back(point);
 }
 
 // Apply velocity scaling for proximity of collisions and singularities.
-// Scale for collisions is read from a shared variable.
-void JogCalcs::applyVelocityScaling(JogArmShared& shared_variables, Eigen::ArrayXd& delta_theta,
-                                    double singularity_scale)
+void JogCalcs::applyVelocityScaling(Eigen::ArrayXd& delta_theta, double singularity_scale)
 {
-  shared_variables.lock();
-  double collision_scale = shared_variables.collision_velocity_scale;
-  shared_variables.unlock();
+  double collision_scale = collision_velocity_scale_;
```

Comment:
I don't think this intermediate variable `collision_scale` is necessary, just use `collision_velocity_scale_`

Reply:
I did this so it couldn't be changed "underneath" this function between tests.  If that happened it would make this function confusing and probably broken.